network/gossip/libp2p/peerManager.go

```diff
@@ -0,0 +1,165 @@
+package libp2p
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/flow/filter"
+)
+
+// Connector connects to peer and disconnects from peer using the underlying networking library
+type Connector interface {
+
+	// ConnectPeers connect to the given flow.Identities and returns a map of identities which failed.
+	// ConnectPeer implementation should be idempotent such that multiple calls to connect to the same peer should not
+	// return an error or create multiple connections
+	ConnectPeers(ctx context.Context, ids flow.IdentityList) map[flow.Identity]error
+
+	// DisconnectPeers disconnect from the given flow.Identities and returns a map of identities which failed.
+	// DisconnectPeers implementation should be idempotent such that multiple calls to connect to the same peer should
+	// not return an error
+	DisconnectPeers(ctx context.Context, ids flow.IdentityList) map[flow.Identity]error
+}
+
+// PeerUpdateInterval is how long the peer manager waits in between attempts to update peer connections
+var PeerUpdateInterval = 1 * time.Minute
+
+// PeerManager adds and removes connections to peers periodically and on request
+type PeerManager struct {
```

Comment:
PeerManager decides the `Who to connect to?` and `When to connect to?` while the Connector provides the `How to Connect to?`